src/test/java/com/crowdin/client/core/http/impl/http/ApacheHttpClientTest.java

```diff
@@ -0,0 +1,224 @@
+package com.crowdin.client.core.http.impl.http;
+
+import com.crowdin.client.core.http.HttpRequestConfig;
+import com.crowdin.client.core.http.JsonTransformer;
+import com.crowdin.client.core.http.exceptions.HttpBadRequestException;
+import com.crowdin.client.core.http.exceptions.HttpException;
+import com.crowdin.client.core.http.impl.json.JacksonJsonTransformer;
+import com.crowdin.client.core.model.BooleanInt;
+import com.crowdin.client.core.model.Credentials;
+import com.crowdin.client.labels.model.Label;
+import org.apache.http.HttpStatus;
+import org.apache.http.HttpVersion;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicStatusLine;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class ApacheHttpClientTest {
+
+    private final Long projectId = 1L;
+    private final Long reportSettingsTemplateId = 2L;
+
+    private final String organizationUrl = "https://testOrganization.api.crowdin.com/api/v2" + "/projects/" + projectId + "/settings-templates/" + reportSettingsTemplateId;
+
+    private CloseableHttpClient client;
+    private CloseableHttpResponse mockedResponse;
+
+    private ApacheHttpClient httpClient;
+
+    private final String json = "{\n" +
+            "  \"id\": 123456789,\n" +
+            "  \"title\": \"Example Label\",\n" +
+            "  \"isSystem\": {\n" +
+            "    \"value\": true\n" +
+            "  }\n" +
+            "}\n";
```

Comment:
Typically, we extract such json test data into separate files in the `resources` directory